doc/features/currencies.rst

```diff
@@ -49,6 +49,35 @@ The Bitcoin currencies provides a single currency: the Bitcoin. It uses XBT as i
     $currencies->contains(new Currency('USD')); // returns boolean whether USD is available in this repository (false)
     $currencies->subunitFor(new Currency('XBT')); // returns the subunit for the Bitcoin (8)
 
+CurrencyList
+-------------
+
+The CurrencyList class provides a way for a developer to build a custom currency repository.
+The class accepts an associative array indexed by the currency code
+and another associative array as the value containing `minorUnit` key and positive integer values.
+
+More specifically it must be an associative array. A currency code key must exists for all currencies and it must be a 
+string. Also `minorUnit` keys must exists and have a positive integer value.
+
+.. code-block:: php
+
+    use Money\Currencies\CurrencyList;
+    use Money\Currency;
+
+    $currencies = new CurrencyList([
+        'MY1' => [
+            'minorUnit' => 2,
```

Comment:
I wonder if we really need this inner associative array. I guess it's there for future options, but for now it's harder to document and understand IMO. Maybe just accept string => integer pairs? We can always add support in a BC way for other types.

WDYT?

Reply:
I agree... I believe it's fine now. Check it out :)